portal/models/questionnaire_response.py

```diff
@@ -428,9 +413,9 @@ def qnrs(self):
             QuestionnaireResponse.status,
             QuestionnaireResponse.document[
                 ('questionnaire', 'reference')].label('instrument_id'),
-            QuestionnaireResponse.authored,
+            QuestionnaireResponse.document['authored'].label('authored'),
             QuestionnaireResponse.encounter_id).order_by(
-            QuestionnaireResponse.authored)
+            QuestionnaireResponse.document['authored'])
```

Comment:
If I understand correctly, this is using the string methods (`__lt__()`/`__gt__()`) for alphabetical sorting.

Are we operating under the assumption that all datetimes are GMT-normalized? Do we need to cast it to a datetime-like object before sorting (so timezones could be normalized)? eg would `2020-11-05T09:15:30-08:00` correctly come before `2020-11-05T08:15:30-05:00`?

Reply:
good point - i don't think it's a real world problem, but best to consider.

Reply:
wow, that was way harder than i expected.  in the end, i found the sort excessively cumbersome and instead, raise an exception if we ever find a situation where the resulting sort order is incorrect.